File: 674-Longest-Continuous-Increasing-Subsequence/findLengthOfLCIS.java
class Solution {
  public int findLengthOfLCIS(int[] nums) {
    int max_length = 0;
    int curr_length = 0;

    for (int i = 0; i < nums.length; i++) {
      curr_length++;
      if (i > 0 && nums[i] <= nums[i-1]) {
        curr_length = 1;
      }
      max_length = Math.max(max_length, curr_length);
    }

    return max_length;
  }
}

File: 674-Longest-Continuous-Increasing-Subsequence/findLengthOfLCIS.py
class Solution:
    def findLengthOfLCIS(self, nums: List[int]) -> int:
        if len(nums) <= 1:
            return len(nums)
        
        max_length = 1
        curr_length = 1
        
        for i in range(1, len(nums)):
            if nums[i] > nums[i-1]:
                curr_length += 1
            else:
                curr_length = 1
            max_length = max(max_length, curr_length)
        
        return max_length